plugins/views/frontend/public/javascripts/countly.models.js

```diff
@@ -1,9 +1,422 @@
 /*global CountlyHelpers, countlyCommon, $, countlySession, jQuery, countlyGlobal, Promise, CV, countlyVue*/
 
-(function() {
-    window.countlyViews = window.countlyViews || {};
-    CountlyHelpers.createMetricModel(window.countlyViews, {name: "views"}, jQuery);
-    var countlyViews = window.countlyViews;
+(function(countlyViews) {
+
+    CountlyHelpers.createMetricModel(countlyViews, {name: "views"}, jQuery);
+
+    countlyViews.service = {
+        fetchData: function(context) {
+            _segment = context.state.selectedSegment;
+            _segmentVal = context.state.selectedSegmentValue;
+            return $.when(countlyViews.initialize());
+        },
+        fetchTotals: function() {
+            var data = {
+                "app_id": countlyCommon.ACTIVE_APP_ID,
+                "method": "views",
+                "period": countlyCommon.getPeriodForAjax(),
+                "action": "getTotals"
+            };
+
+            return new Promise(function(resolve, reject) {
```

Comment:
You don't need to wrap CV.$.ajax with native Promise, it is already wrapped with one. You can just do `return CV.$.ajax(...).then(...).catch(...)`